From b9affdd8bf5dbe302752725c9321c6cf94b165c5 Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Mon, 6 Jan 2014 16:50:30 -0800
Subject: [PATCH 2712/5965] Add vf NULL pointer check when display side
 successfully peeked a frame but there is NULL frame pointer returned later
 for get method when a decoder is unregistered between.

---
 drivers/amlogic/amports/video.c | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index 68d988e5454c..aa9bcf9cf54b 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -1944,12 +1944,12 @@ static irqreturn_t vsync_isr(int irq, void *dev_id)
     if ((!vf) && cur_dispbuf && (video_property_changed)) {
         vsync_toggle_frame(cur_dispbuf);
     }
+
     if (!vf) {
         underflow++;
     }
 
     while (vf) {
-		//timer_count = 0 ;
         if (vpts_expire(cur_dispbuf, vf)
 #ifdef INTERLACE_FIELD_MATCH_PROCESS
             || interlace_field_type_match(vout_type, vf)
@@ -1963,16 +1963,14 @@ static irqreturn_t vsync_isr(int irq, void *dev_id)
                        timestamp_pcrscr_get());
 
             amlog_mask_if(toggle_cnt > 0, LOG_MASK_FRAMESKIP, "skipped\n");
+
 #if defined(CONFIG_AM_VECM)
-			ve_on_vs(vf);
-#endif
-#if 0 //def CONFIG_VSYNC_RDMA
-            if(dispbuf_to_put) {
-                video_vf_put(dispbuf_to_put);
-                dispbuf_to_put = NULL;
-            }
+            ve_on_vs(vf);
 #endif
+
             vf = video_vf_get();
+            if (!vf) break;
+
             use_prot = get_use_prot();
             if (use_prot && (video_prot.video_started || video_prot.src_vframe_width != vf->width || video_prot.src_vframe_height != vf->height)) {
                 video_prot_init(&video_prot, vf);
@@ -2011,13 +2009,13 @@ static irqreturn_t vsync_isr(int irq, void *dev_id)
             frame_repeat_count = 0;
 #endif
             vf = video_vf_peek();
-		if (!vf) {
-			next_peek_underflow++;
-		}
+            if (!vf) {
+                next_peek_underflow++;
+            }
 
-		        if(debug_flag & DEBUG_FLAG_TOGGLE_FRAME_PER_VSYNC){
-		            break;
-		        }
+           if (debug_flag & DEBUG_FLAG_TOGGLE_FRAME_PER_VSYNC) {
+               break;
+           }
         } else {
 #ifdef SLOW_SYNC_REPEAT
             /* check if current frame's duration has expired, in this example
@@ -2040,6 +2038,8 @@ static irqreturn_t vsync_isr(int irq, void *dev_id)
                 }
 #endif
                 vf = video_vf_get();
+                if (!vf) break;
+
                 use_prot = get_use_prot();
                 if (use_prot && (video_prot.video_started || video_prot.src_vframe_width != vf->width || video_prot.src_vframe_height != vf->height)) {
                     video_prot_init(&video_prot, vf);
-- 
2.19.0

